src/Aspire.Hosting/Exec/ExecResourceManager.cs

```diff
@@ -0,0 +1,213 @@
+        await _resourceNotificationService.WaitForResourceAsync(execResource.Name, targetState: KnownResourceStates.Starting, cancellationToken).ConfigureAwait(false);
+        var dcpExecResourceName = GetDcpExecResourceName(execResource);
+
+        yield return new CommandOutput
+        {
+            Text = $"Aspire exec starting...",
+            Type = type
+        };
+
+        // in the background wait for the exec resource to be running to change log type
+        _ = Task.Run(async () =>
+        {
+            await _resourceNotificationService.WaitForResourceAsync(execResource.Name, targetState: KnownResourceStates.Running, cancellationToken).ConfigureAwait(false);
+            type = "running";
+        }, cancellationToken);
+
+        // in the background wait for the exec resource to reach terminal state. Once done we can complete logging
+        _ = Task.Run(async () =>
+        {
+            await _resourceNotificationService.WaitForResourceAsync(execResource.Name, targetStates: KnownResourceStates.TerminalStates, cancellationToken).ConfigureAwait(false);
+            _resourceLoggerService.Complete(dcpExecResourceName); // complete stops the `WatchAsync` async-foreach below
+        }, cancellationToken);
+
+        await foreach (var logs in _resourceLoggerService.WatchAsync(dcpExecResourceName).WithCancellation(cancellationToken).ConfigureAwait(false))
+        {
+            foreach (var log in logs)
+            {
+                yield return new CommandOutput
+                {
+                    Text = log.Content,
+                    IsErrorMessage = log.IsErrorMessage,
+                    LineNumber = log.LineNumber,
+                    Type = type
+                };
+            }
+        }
+    }
+
+    public IResource? ConfigureExecResource()
+    {
+        if (!_execOptions.Enabled)
+        {
+            return null;
+        }
+
+        var targetResource = _model.Resources.FirstOrDefault(x => x.Name.Equals(_execOptions.ResourceName, StringComparisons.ResourceName));
+        if (targetResource is null)
+        {
+            _logger.LogError("Exec resource with name '{ResourceName}' not found in the model resources.", _execOptions.ResourceName);
+            throw new InvalidOperationException($"Exec resource with name '{_execOptions.ResourceName}' not found in the model resources.");
+        }
+
+        var execResource = BuildResource(targetResource);
+
+        _logger.LogInformation("Resource '{ResourceName}' has been successfully built and added to the model resources.", execResource.Name);
+        _execResourceInitialized.SetResult(execResource);
+        return execResource;
+    }
+
+    IResource BuildResource(IResource targetExecResource)
+    {
+        return targetExecResource switch
+        {
+            ProjectResource prj => BuildAgainstProjectResource(prj),
+            _ => throw new InvalidOperationException($"Target resource does not support exec mode.")
```

Comment:
```suggestion
            _ => throw new InvalidOperationException("Target resource does not support exec mode.")
```

Reply:
changed, now has the interpolated string